sale_automatic_workflow/models/stock_picking.py

```diff
@@ -4,20 +4,33 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from odoo import api, fields, models
+from odoo.tools import float_compare
 
 
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
     workflow_process_id = fields.Many2one(
-        comodel_name='sale.workflow.process',
-        string='Sale Workflow Process'
+        comodel_name="sale.workflow.process", string="Sale Workflow Process"
     )
 
     @api.multi
     def validate_picking(self):
         for picking in self:
-            picking.force_assign()
-            self.env['stock.immediate.transfer'].create(
-                {'pick_ids': [(4, picking.id)]}).process()
+            picking.action_assign()
+            for move in picking.move_lines:
+                rounding = move.product_id.uom_id.rounding
+                if (
+                    float_compare(
+                        move.quantity_done,
+                        move.product_qty,
+                        precision_rounding=rounding,
+                    )
+                    == -1
+                ):
+                    move.quantity_done = move.product_qty
```

Comment:
may be just one .update() to move ?

Reply:
There is only this field to update, it is fine like this (I removed the `move.state = 'assigned' line in my own commit).